test/wallet/keys/hd_private.cpp

```diff
@@ -128,4 +128,112 @@ BOOST_AUTO_TEST_CASE(hd_private__derive_public__long_seed__expected)
     BOOST_REQUIRE_EQUAL(m0xH1yH2_pub.encoded(), "xpub6FnCn6nSzZAw5Tw7cgR9bi15UV96gLZhjDstkXXxvCLsUXBGXPdSnLFbdpq8p9HmGsApME5hQTZ3emM2rnY5agb9rXpVGyy3bdW6EEgAtqt");
 }
 
+BOOST_AUTO_TEST_CASE(hd_private__constructor__null_key_decodes_to_invalid__expected)
```

Comment:
This:

`hd_private__constructor__null_key_decodes_to_invalid__expected`

would be better as:

`hd_private__constructor__null_key__decodes_to_invalid`

Others as well. Third part is the thing being tested and fourth part is the expectation.